javav2/example_code/support/src/main/java/com/example/support/SupportScenario.java

```diff
@@ -0,0 +1,377 @@
+                .build();
+
+            DescribeCommunicationsResponse response = supportClient.describeCommunications(communicationsRequest);
+            List<Communication> communications = response.communications();
+            for (Communication comm: communications) {
+                System.out.println("the body is: " + comm.body());
+
+                //Get the attachment id value.
+                List<AttachmentDetails> attachments = comm.attachmentSet();
+                for (AttachmentDetails detail : attachments) {
+                    attachId = detail.attachmentId();
+                }
+            }
+            return attachId;
+
+        } catch (SupportException e) {
+            System.out.println(e.getLocalizedMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+
+    public static void addAttachSupportCase(SupportClient supportClient, String caseId, String attachmentSetId) {
+        try {
+            AddCommunicationToCaseRequest caseRequest = AddCommunicationToCaseRequest.builder()
+                .caseId(caseId)
+                .attachmentSetId(attachmentSetId)
+                .communicationBody("Please refer to attachment for details.")
+                .build();
+
+            AddCommunicationToCaseResponse response = supportClient.addCommunicationToCase(caseRequest);
+            if (response.result())
+                System.out.println("You have successfully added a communication to an Amazon Web Services Support case");
+            else
+                System.out.println("There was an error adding the communication to an Amazon Web Services Support case");
+
+        } catch (SupportException e) {
+            System.out.println(e.getLocalizedMessage());
+            System.exit(1);
+        }
+   }
+
+    public static String addAttachment(SupportClient supportClient, String fileAttachment) {
+        try {
+            File myFile = new File(fileAttachment);
+            InputStream sourceStream = new FileInputStream(myFile);
+            SdkBytes sourceBytes = SdkBytes.fromInputStream(sourceStream);
+
+            Attachment attachment = Attachment.builder()
+                .fileName(myFile.getName())
+                .data(sourceBytes)
+                .build();
+
+            AddAttachmentsToSetRequest setRequest = AddAttachmentsToSetRequest.builder()
+                .attachments(attachment)
+                .build();
+
+            AddAttachmentsToSetResponse response = supportClient.addAttachmentsToSet(setRequest);
+            return response.attachmentSetId();
+
+        } catch (SupportException | FileNotFoundException e) {
+            System.out.println(e.getLocalizedMessage());
+            System.exit(1);
+        }
+        return "";
+    }
+
+    public static void getOpenCase(SupportClient supportClient) {
+        try {
+            DescribeCasesRequest describeCasesRequest = DescribeCasesRequest.builder()
+                .maxResults(20)

```

Comment:
This step specifies getting open cases for the current day. It doesn't look like this request has a date filter like the one for resolved cases.